rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -295,18 +297,22 @@ void SequentialWriter::write(std::shared_ptr<rosbag2_storage::SerializedBagMessa
   const auto message_timestamp = std::chrono::time_point<std::chrono::high_resolution_clock>(
     std::chrono::nanoseconds(message->time_stamp));
 
-  if (should_split_bagfile(message_timestamp)) {
-    split_bagfile();
+  if (!is_first_message_) {
```

Comment:
I feel like this logic should be flipped. As it is now, it reads like "If this isn't the first message, update the starting time, and say that every future message is the first message".

I feel like this could be a source of confusion to future maintainers.

Reply:
@Kaju-Bubanja I agree with @jhdcs.
Could you please init `is_first_message_{true}` check `if (is_first_message_)` here and set it to **false** afterwards?

Reply:
Right, that makes more sense :D Reviews are important^^